packages/dds/tree/src/feature-libraries/editable-tree/editableTreeContext.ts

```diff
@@ -219,55 +212,67 @@ export class ProxyContext implements EditableTreeContext {
 		return this.forest.schema;
 	}
 
-	public setNodeValue(path: UpPath, value: Value): boolean {
-		return this.runTransaction((editor) => editor.setValue(path, value));
+	public setNodeValue(path: UpPath, value: Value): void {
+		assert(
+			this.editor !== undefined,
+			0x45a /* an editor is required to edit the EditableTree */,
+		);
+		this.editor.setValue(path, value);
 	}
 
 	public setValueField(
 		path: UpPath | undefined,
 		fieldKey: FieldKey,
 		newContent: ITreeCursor,
-	): boolean {
-		return this.runTransaction((editor) => {
-			const field = editor.valueField(path, fieldKey);
-			field.set(newContent);
-		});
+	): void {
+		assert(
+			this.editor !== undefined,
+			0x45a /* an editor is required to edit the EditableTree */,
```

Comment:
you can not have the same assert ID in multiple locations. That will fail policy check. THis number is suppose to identify a unique location in the codebase. You can use the string message in all these places and let the tagger tool generate numbers for the next time we run it, or avoid having all these checks in the first place.

I think the only place that currently doesn't provide an editor is some tests. We can make those work by providing a mock implementation that errors if you call any of its editing functions (or just cast {} to editor or something). I think this would be cleaner.

Another approach would be to have a getter for editor with the assert in it.

Reply:
Oh. I always assumed that if the string matched, then the error number would match. But I guess it makes sense to have duplicates if the point is to identify a specific line of code.

I like the mock idea too, I'll do that.